artiq/coredevice/comm_kernel.py

```diff
@@ -651,6 +652,12 @@ def read_exception_string():
             message = read_exception_string()
             params = [self._read_int64() for _ in range(3)]
 
+            if any(err in name for err in ["RTIOUnderflow",
+                    "RTIOOverflow", "RTIODestinationUnreachable"]):
```

Comment:
Why can't you do this later and use ``python_exn_type`` instead of duplicating, poorly, the exception decoding logic?